File: ontrack-extension-svn/src/main/java/net/nemerosa/ontrack/extension/svn/property/SVNProjectConfigurationProperty.java
package net.nemerosa.ontrack.extension.svn.property;

import lombok.Data;
import net.nemerosa.ontrack.extension.svn.SVNConfiguration;

/**
 * Associates a {@link net.nemerosa.ontrack.model.structure.Project} with a
 * {@link net.nemerosa.ontrack.extension.svn.SVNConfiguration}.
 */
@Data
public class SVNProjectConfigurationProperty {

    /**
     * Link to the SVN configuration
     */
    private final SVNConfiguration configuration;

    /**
     * Path of the main project branch (trunk) in this configuration. The path is relative to the root
     * of the repository.
     */
    private final String projectPath;

    /**
     * Derived property: the full URL to the Subversion URL.
     */
    public String getUrl() {
        return configuration.getUrl(projectPath);
    }

}

File: ontrack-extension-svn/src/main/java/net/nemerosa/ontrack/extension/svn/property/SVNProjectConfigurationProperty.kt
package net.nemerosa.ontrack.extension.svn.property

import net.nemerosa.ontrack.extension.svn.model.SVNConfiguration
import net.nemerosa.ontrack.model.support.ConfigurationProperty

/**
 * Associates a [net.nemerosa.ontrack.model.structure.Project] with a
 * [net.nemerosa.ontrack.extension.svn.model.SVNConfiguration].
 *
 * @property configuration Link to the SVN configuration
 * @property projectPath Path of the main project branch (trunk) in this configuration. The path is relative to the root
 * of the repository.
 */
class SVNProjectConfigurationProperty(
        private val configuration: SVNConfiguration,
        val projectPath: String
) : ConfigurationProperty<SVNConfiguration> {

    override fun getConfiguration(): SVNConfiguration = configuration

    /**
     * Derived property: the full URL to the Subversion URL.
     */
    val url: String
        get() = configuration.getUrl(projectPath)

}
